From e3ed764bb246e312050416f94b1e04a1569d1a5a Mon Sep 17 00:00:00 2001
In-Reply-To: <529de1a6-7898-4f6a-bda5-8b12b11cb987@ovn.org>
References: <529de1a6-7898-4f6a-bda5-8b12b11cb987@ovn.org>
From: Frode Nordahl <fnordahl@ubuntu.com>
Date: Mon, 8 Apr 2024 15:47:18 +0200
Subject: [PATCH v2] tests: Fix compatibility issue with Python 3.13 in
 vlog.at.
To: dev@openvswitch.org

The vlog - Python3 test makes use of output from Python
Tracebacks in its test assertion.

In Python 3.13 a line with tophat (``^``) markers is added below
Tracebacks from calls to assert [0], which makes the test fail.
This change of behavior is also backported to the Python 3.12 and
3.11 stable branches [1].

Strip lines containing one or more occurence of the ``^``
character from the output before performing the test assertions.

0: https://github.com/python/cpython/pull/105935
1: https://github.com/python/cpython/issues/116034
Reported-at: https://launchpad.net/bugs/2060434
Signed-off-by: Frode Nordahl <fnordahl@ubuntu.com>
---
 tests/vlog.at | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tests/vlog.at b/tests/vlog.at
index 785014956..efe91479a 100644
--- a/tests/vlog.at
+++ b/tests/vlog.at
@@ -8,6 +8,7 @@ AT_CHECK([$PYTHON3 $srcdir/test-vlog.py --log-file log_file \
 
 AT_CHECK([sed -e 's/.*-.*-.*T..:..:..Z |//' \
 -e 's/File ".*", line [[0-9]][[0-9]]*,/File <name>, line <number>,/' \
+-e '/\^\+/d' \
 stderr_log], [0], [dnl
   0  | module_0 | EMER | emergency
   1  | module_0 | ERR | error
-- 
2.43.0

